operator/cnp_event.go

```diff
@@ -68,19 +68,19 @@ func enableCNPWatcher(apiextensionsK8sClient apiextensionsclientset.Interface) e
 	}
 
 	if kvstoreEnabled() {
+		cnpStatusMgr = k8s.NewCNPStatusEventHandler(nil, cnpStore, cnpStatusUpdateInterval)
 		cnpSharedStore, err := store.JoinSharedStore(store.Configuration{
 			Prefix: k8s.CNPStatusesPath,
 			KeyCreator: func() store.Key {
 				return &k8s.CNPNSWithMeta{}
 			},
+			Observer: cnpStatusMgr,
 		})
 		if err != nil {
 			return err
 		}
 
-		cnpStatusMgr = k8s.NewCNPStatusEventHandler(cnpSharedStore, cnpStore, cnpStatusUpdateInterval)
-
-		go cnpStatusMgr.WatchForCNPStatusEvents()
+		cnpStatusMgr.UpdateCNPStore(cnpSharedStore)
```

Comment:
Please add a comment that is safe to update the store here given that the store will only be used by the `StartStatusHandler` function. (Same for the CCNP similar code)